crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeFactory.kt

```diff
@@ -0,0 +1,36 @@
+package net.corda.v5.crypto.merkle
+
+import net.corda.v5.crypto.DigestAlgorithmName
+
+/**
+ * [MerkleTreeFactory] creates [MerkleTree]s, [MerkleProof]s and [MerkleTreeHashDigestProvider].
+ */
+interface MerkleTreeFactory {
```

Comment:
I think we should add some constants to this for the various 'standard' digest algorithms and also for the parameter keys that they require

Reply:
I think this was initially `MerkleService` or something. I've added `@DoNotImplement` to the interface and `@Suspendable` to the functions -- is that what you mean?
The implementation class in corda-runtime-os is wired up with `@Component` and whatnot.

Can you elaborate about the constants?

Reply:
@mnesbit 
> I think we should add some constants to this for the various 'standard' digest algorithms and also for the parameter keys that they require

I have pushed some 1-1 changes and introduced MerkleTreeHashDigestProviderOption and MerkleTreeHashDigestProviderName
Is this what you meant?